jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -205,35 +216,55 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
         """
         spectral_cube = self._app._jdaviz_helper._loaded_flux_cube
         uncert_cube = self._app._jdaviz_helper._loaded_uncert_cube
+        uncertainties = None
 
         # This plugin collapses over the *spatial axes* (optionally over a spatial subset,
         # defaults to ``No Subset``). Since the Cubeviz parser puts the fluxes
         # and uncertainties in different glue Data objects, we translate the spectral
         # cube and its uncertainties into separate NDDataArrays, then combine them:
-        if self.aperture.selected != self.aperture.default_text:
+        if self.aperture.selected != self.aperture.default_text and self.wavelength_dependent:
             nddata = spectral_cube.get_subset_object(
                 subset_id=self.aperture.selected, cls=NDDataArray
             )
-            uncertainties = uncert_cube.get_subset_object(
-                subset_id=self.aperture.selected, cls=StdDevUncertainty
+            if uncert_cube:
+                uncertainties = uncert_cube.get_subset_object(
+                    subset_id=self.aperture.selected, cls=StdDevUncertainty
+                )
+            # Exact slice mask of cone aperture through the cube
+            cone_mask = self.cone_aperture()
+            if self.aperture_method_selected.lower() == 'center':
+                flux = nddata.data << nddata.unit
+            else:
+                # Apply the fractional pixel array to the flux cube
+                flux = (cone_mask * nddata.data) << nddata.unit
```

Comment:
Sounds good, I'll change that. Can `min` and `max` use the same code as `sum` or does that need to be altered as well?

Reply:
I would take the min/max of the pixels that are fully within the aperture. Since the min/max stats are useful in edge cases, I don't think partial pixel logic needs to be thought about too deeply in that case.